src/Resources/contao/library/Contao/InsertTags.php

```diff
@@ -52,15 +52,20 @@ public function replace($strBuffer, $blnCache=true)
 			return \StringUtil::restoreBasicEntities($strBuffer);
 		}
 
-		$tags = preg_split('/{{(([^{}]*|(?R))*)}}/', $strBuffer, -1, PREG_SPLIT_DELIM_CAPTURE);
+		$tags = preg_split('/{{([^{}]+)}}/', $strBuffer, -1, PREG_SPLIT_DELIM_CAPTURE);
+
+		if (count($tags) < 2)
+		{
+			return \StringUtil::restoreBasicEntities($strBuffer);
```

Comment:
@Toflar This new return statement actually prevents endless loops now, so we do no longer need the nesting level counter.